src/System.Management.Automation/engine/CompiledCommandParameter.cs

```diff
@@ -426,91 +437,84 @@ internal IEnumerable<ParameterSetSpecificMetadata> GetMatchingParameterSetData(u
-                }
+                case ValidateArgumentsAttribute validateAttr:
+                    if (validationAttributes == null)
+                    {
+                        validationAttributes = new Collection<ValidateArgumentsAttribute>();
+                    }
 
-                return;
-            }
+                    validationAttributes.Add(validateAttr);
+                    if ((attribute is ValidateNotNullAttribute) || (attribute is ValidateNotNullOrEmptyAttribute))
+                    {
+                        this.CannotBeNull = true;
+                    }
 
-            AliasAttribute aliasAttr = attribute as AliasAttribute;
-            if (aliasAttr != null)
-            {
-                if (aliases == null)
-                {
-                    aliases = aliasAttr.aliasNames;
-                }
-                else
-                {
-                    var prevAliasNames = aliases;
-                    var newAliasNames = aliasAttr.aliasNames;
-                    aliases = new string[prevAliasNames.Length + newAliasNames.Length];
-                    Array.Copy(prevAliasNames, aliases, prevAliasNames.Length);
-                    Array.Copy(newAliasNames, 0, aliases, prevAliasNames.Length, newAliasNames.Length);
-                }
+                    return;
 
-                return;
-            }
+                case AliasAttribute aliasAttr:
+                    if (aliases == null)
+                    {
+                        aliases = aliasAttr.aliasNames;
+                    }
+                    else
+                    {
+                        var prevAliasNames = aliases;
+                        var newAliasNames = aliasAttr.aliasNames;
+                        aliases = new string[prevAliasNames.Length + newAliasNames.Length];
+                        Array.Copy(prevAliasNames, aliases, prevAliasNames.Length);
+                        Array.Copy(newAliasNames, 0, aliases, prevAliasNames.Length, newAliasNames.Length);
+                    }
 
-            ArgumentTransformationAttribute argumentAttr = attribute as ArgumentTransformationAttribute;
-            if (argumentAttr != null)
-            {
-                if (argTransformationAttributes == null)
-                    argTransformationAttributes = new Collection<ArgumentTransformationAttribute>();
-                argTransformationAttributes.Add(argumentAttr);
-                return;
-            }
+                    return;
 
-            AllowNullAttribute allowNullAttribute = attribute as AllowNullAttribute;
-            if (allowNullAttribute != null)
-            {
-                this.AllowsNullArgument = true;
-                return;
-            }
+                case ArgumentTransformationAttribute argumentAttr:
+                    if (argTransformationAttributes == null)
+                    {
+                        argTransformationAttributes = new Collection<ArgumentTransformationAttribute>();
```

Comment:
Maybe use `??=`?

Reply:
Good idea. Much cleaner. Will be in next commit.